app/email.go

```diff
@@ -589,3 +589,47 @@ func (es *EmailService) CreateVerifyEmailToken(userId string, newEmail string) (
 
 	return token, nil
 }
+
+func (es *EmailService) SendAtUserLimitWarningEmail(email string, locale string, siteURL string) (bool, *model.AppError) {
```

Comment:
Given that is the same template but with different data we can probably we can reuse most of the logic, create a `sendUserLimitEmail` and then call it from these two methods with the proper data. What do you think?

Reply:
I've done it this way for a few reasons:
1. I think it's more readable when the code calls `SendAtUserLimitWarningEmail` instead of deriving which type of email it is from the arguments
2. It seems like this sort of pattern is followed elsewhere in the file, see `sendChangeUsernameEmail` and `sendEmailChangeEmail` which both use the same template
3. We'd need 6-7 function arguments, and I'm not a fan of huge definitions like that

I'm swaying to keeping it as is - if you really want it changed, let me know!

Reply:
No hard opinion here 2/5 so if you think that's ok, works for me